Updating Docs for 2.1 Release (#164)
SuperDARN/pydarn: docs/user/coordinates.md

<!--Copyright (C) SuperDARN Canada, University of Saskatchewan 
Author(s): Marina Schmidt 
Modifications:

Disclaimer:
pyDARN is under the LGPL v3 license found in the root directory LICENSE.md 
Everyone is permitted to copy and distribute verbatim copies of this license 
document, but changing it is not allowed.

This version of the GNU Lesser General Public License incorporates the terms
and conditions of version 3 of the GNU General Public License, supplemented by
the additional permissions listed below.
-->

# Coordinate Systems 
---

> pyDARN offers several different coordinate systems for plotting: 

Reviewer: ```suggestion
pyDARN offers several different coordinate systems and units for plotting: 
```

## Range-Time Plots 

> **Range Gates**: `Coords.RANGE_GATE` a rectangle section determined by beam width and set distance for each range (nominally 45 km). RAWACF and FITACF data give their parameter values with respect to range gates  

Reviewer: ```suggestion
**Range Gates**: `Coords.RANGE_GATE` a rectangle section determined by beam width and set distance for each range (nominally 45 km). RAWACF and FITACF data give their parameter values with respect to range gates. Range gates are a unit-less measure of estimating distance.
```

> **Slant Range**: `Coords.SLANT_RANGE` is a conversion from range gates to km units.  

Reviewer: ```suggestion
**Slant Range**: `Coords.SLANT_RANGE` is a conversion from range gates to km units.  Slant range estimates the distance of ionospheric echos from the radar, using the time it takes for the radio wave to travel to the ionosphere and return, assuming the radio wave is travelling at the speed of light.
```

Author: Do we want to add a warning with rxrise and frang?

Reviewer: Yeah, I'm just looking over those issues I made to write it properly

Reviewer: I've amended the text to include a run down of things that are of note in how we calculate slant range, but that it's outside our remit to fix these and were essentially waiting it out until it's fixed at a higher level.

I don't know if the formatting is right in the suggestion though, so it might need going in and fixing?

> **Ground Scatter-Mappped Range**: `Coords.GROUND_SCATTER_MAPPED_RANGE` uses echos from groundscatter to adjust slant-range coordinates to be more accurate (km) based on [Dr. Bill Bristow's paper](https://doi.org/10.1029/93JA01470). Implemented by Dr. Nathaniel Frissell and Francis Tholley from University of Scranton. 

Reviewer: ```suggestion
**Ground Scatter Mapped Range**: `Coords.GROUND_SCATTER_MAPPED_RANGE` uses echos from ground scatter to adjust slant-range coordinates to be more accurate based on [Dr. Bill Bristow's paper](https://doi.org/10.1029/93JA01470). Implemented by Dr. Nathaniel Frissell and Francis Tholley from University of Scranton. Measured in km.
```

> **Geographic** (Coming Soon)
> 
> **AACGM** (Coming Soon)

Reviewer: I dont think either of these apply to RTP plots. 
```suggestion
```

Author: Apparently, you can get them DaVitpy and RST provide them so I thought we could as well

Reviewer: Aah, right so you plot the range gates geographic or geomagnetic longitude/latitude up the y-axis along the beam you're interested in? I get the idea, but it's kinda only giving you half of the information about the position of the echoes, but I guess it's useful to someone.

Reviewer: I've amended the suggestion accordingly :P

## Geographic Plots

Geographic plots include: fan, grid and convection map (coming soon) plots

**Geographic**: `Coords.GEOGRAPHIC` is the standard geographical coordinate system for latitude and longitude (degree) (coming soon)

**AACGM**: `Coords.AACGM` is [Altitude Adjusted Corrected Geogmagnetic Coordinates developed by Dr. Simon Shepherd](http://superdarn.thayer.dartmouth.edu/aacgm.html) are an extension of corrected geomagnetic (CGM) coordinates that more accurately represent the actual magnetic field. In AACGM coordinates points along a given magnetic field line are given the same coordinates and thus are a better reflection of magnetic conjugacy. pyDARN uses AACGM-V2 from the [aacgmv2 python library](https://pypi.org/project/aacgmv2/). Implemented by Dr. Daniel Billett from University of Saskatchewan. 

**Ground Scatter Mapped Geographic** (Coming Soon)

> **Ground Scatter Mapped Geographic** (Coming Soon)

Reviewer: ```suggestion
**Ground Scatter Mapped Geomagnetic** (Coming Soon)
```
?
these two will be plotted on a geographic or aacgm axes respectively, but the data or fov or whatever will have different values in those coordinate systems - so they're more like an option for plotting unit rather than coordinate system, if you get me.

Author: well, we use slant range to guess the geographic location which is then used for AACGM. So this would be used GSRM instead of slant range so maybe not a coordinate system but something we would pass into plots

Reviewer: yeah, so its the location of the data that changes using the GSMR, but the axes will still show geographic and geomagnetic lats/lons

Also sorry, this suggestion was completely unrelated to the comment after it.

Author: Okay no worries, I think this is still important. So if they are not coordinates? what would you classify it as? Be nice to have it here.

Reviewer: I think a while ago I suggested just calling them units (but I mean they're technically a coordinate right, but they're not the coordinate system itself?) Like you can mix and match which units you want to plot in which coordinate systems you want to view it on - for the fan/fov/grid plots anyway. The 'coordinate system' for the RTP plots is not really a thing, that's just plotting the unit up the y-axis against time, but the y-axis can have many different units, that are like the geographic latitude, etc etc.
